packages/transform/src/lib/TransformMiddleware.ts

```diff
@@ -0,0 +1,485 @@
+    const { after } = this.context;
+    const hook = async (value: AfterValue, key: string) => {
+      return payload.hook((await after(value, key, null)) as ReturnValue, key) as ReturnValue;
+    };
+
+    await this.provider[Method.Each]({ ...payload, hook });
+
+    return payload;
+  }
+
+  @PreProvider()
+  public async [Method.Ensure]<ReturnValue = BeforeValue, StoredValue = AfterValue>(
+    payload: Payload.Ensure<ReturnValue>
+  ): Promise<Payload.Ensure<StoredValue>> {
+    const { key, defaultValue } = payload;
+    const { before, autoTransform } = this.context;
+
+    payload.defaultValue = (await before(defaultValue as unknown as BeforeValue, key, null)) as ReturnValue;
+
+    await this.provider[Method.Ensure](payload as unknown as Payload.Ensure<AfterValue>);
+
+    if (!(await this.isTransformed(key))) {
+      if (autoTransform === true) {
+        const { data } = await this.provider[Method.Get]({ method: Method.Get, errors: [], key, path: [] });
+
+        await this.provider[Method.Set]({ method: Method.Set, errors: [], key, path: [], value: data });
+        await this.updateMetadataPath(key, this.objectPathKeys(data));
+      }
+
+      process.emitWarning(
+        `The ensured data at "${key}" has not been transformed yet, please enable "autoTransform" to transform the data automatically or set() the data at the key to transform it.`
+      );
+    }
+
+    return payload as unknown as Payload.Ensure<StoredValue>;
+  }
+
+  @PreProvider()
+  public async [Method.Entries]<StoredValue = AfterValue, ReturnValue = BeforeValue>(
+    payload: Payload.Entries<StoredValue>
+  ): Promise<Payload.Entries<ReturnValue>> {
+    const { after } = this.context;
+    const { data } = await this.provider[Method.Entries]({ method: Method.Entries, errors: [] });
+
+    payload.data = {};
+
+    for (const [key, value] of Object.entries(data!)) {
+      payload.data[key] = (await after(value as AfterValue, key, null)) as StoredValue;
+    }
+
+    return payload as unknown as Payload.Entries<ReturnValue>;
+  }
+
+  public async [Method.Map]<StoredValue = AfterValue, ReturnValue = BeforeValue>(
+    payload: Payload.Map.ByHook<StoredValue, ReturnValue>
+  ): Promise<Payload.Map.ByHook<StoredValue, ReturnValue>>;
+
+  public async [Method.Map]<ReturnValue = BeforeValue>(payload: Payload.Map.ByPath<ReturnValue>): Promise<Payload.Map.ByPath<ReturnValue>>;
+
+  @PreProvider()
+  public async [Method.Map]<StoredValue = AfterValue, ReturnValue = BeforeValue>(
+    payload: Payload.Map<StoredValue, ReturnValue>
+  ): Promise<Payload.Map<StoredValue, ReturnValue>> {
+    const { after } = this.context;
+
+    if (isMapByHookPayload(payload)) {
+      const hook = async (value: AfterValue, key: string) => {
+        // @ts-expect-error 2322 - We know that the hook is defined.
+        return payload.hook!((await after(value, key, null)) as StoredValue, key);
```

Comment:
Destructure the hook like so:
```ts
const { hook } = payload
```
inside the isMapByHookPayload, to fix this error, and other places where this occurs